tests/integration_test/managed_service_config_test.go

```diff
@@ -102,3 +105,53 @@ func TestManagedServiceConfig(t *testing.T) {
 		}
 	}
 }
+
+type configsHandler struct {
+	m                *comp.MockServiceMrg
+	failWith429Times int
+	curFailCnt       int
+}
+
+func (h *configsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if h.curFailCnt < h.failWith429Times {
+		h.curFailCnt += 1
+		w.WriteHeader(http.StatusTooManyRequests)
+		return
+	}
+
+	serviceConfigByte, _ := proto.Marshal(h.m.ServiceConfig)
+	h.m.LastServiceConfig = serviceConfigByte
+	_, _ = w.Write(serviceConfigByte)
+}
+
+func TestRetryCallServiceManagement(t *testing.T) {
+	t.Parallel()
+
+	configID := "test-config-id"
+
+	args := []string{"--service_config_id=" + configID,
+		"--rollout_strategy=fixed", "--healthz=/healthz"}
+
+	s := env.NewTestEnv(comp.TestRetryCallServiceManagement, platform.EchoSidecar)
+	defer s.TearDown(t)
+
+	s.MockServiceManagementServer.ConfigsHandler = &configsHandler{
+		m:                s.MockServiceManagementServer,
+		failWith429Times: 2,
+	}
+
+	if err := s.Setup(args); err != nil {
+		t.Fatalf("fail to setup test env, %v", err)
+	}
+
+	url := fmt.Sprintf("http://localhost:%v/echo", s.Ports().ListenerPort)
+
+	resp, err := echoClient.DoPost(fmt.Sprintf("%s?key=api-key", url), echo)
+	if err != nil {
+		t.Errorf("got unexpected error: %v", err)
+	}
+	wantResp := `{"message":"hello"}`
```

Comment:
can we have two test cases, one fail and one success?

Reply:
why could n't we have a negative test case?

Reply:
Because integration test use the binary and we cannot modify global map, we have to wait 1min30sec to see the failure.